[Java][CI] Testing cookbooks on Java fails to import extension javadoctest from sphinx

Java nightly builds have been failing for some days with:
https://github.com/apache/arrow-cookbook/actions/runs/3852256069/jobs/6564227455
https://github.com/apache/arrow-cookbook/actions/runs/3879814786/jobs/6617345612
```
>>> Testing Java Cookbook <<<

cd java && make javadoctest
Running Sphinx v6.1.1

Running with Arrow version: 11.0.0-SNAPSHOT
Extension error:
Could not import extension javadoctest (exception: cannot import name 'Dict' from 'sphinx.ext.doctest' (/Users/runner/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/sphinx/ext/doctest.py))
make[1]: *** [javadoctest] Error 2
make: *** [javatest] Error 2
Error: Process completed with exit code 2.
```

--- a/java/ext/javadoctest.py
+++ b/java/ext/javadoctest.py
@@ -17,8 +17,9 @@
 import os
 import pathlib
 import subprocess
+from typing import Any, Dict
 
-from sphinx.ext.doctest import (Any, DocTestBuilder, TestcodeDirective,
+from sphinx.ext.doctest import (DocTestBuilder, TestcodeDirective,
                                 TestoutputDirective, doctest, sphinx)
 from sphinx.locale import __
 
@@ -99,7 +100,7 @@ class JavaDocTestBuilder(DocTestBuilder):
         output = (4*' ').join(output.split('\t'))
         return output
 
-def setup(app):
+def setup(app) -> Dict[str, Any]:
     app.add_directive("testcode", JavaTestcodeDirective)
     app.add_directive("testoutput", TestoutputDirective)
     app.add_builder(JavaDocTestBuilder)

